set.mm

```diff
@@ -660473,6 +660392,215 @@ D Fn ( I ... ( M - 1 ) ) /\
+    rhmcomulmpl.i $e |- ( ph -> I e. V ) $.
+    rhmcomulmpl.h $e |- ( ph -> H e. ( R RingHom S ) ) $.
+    rhmcomulmpl.f $e |- ( ph -> F e. B ) $.
+    rhmcomulmpl.g $e |- ( ph -> G e. B ) $.
+    $( The ring homomorphism in ~ rhmmpl preserves multiplication.
+       (Contributed by SN, 8-Feb-2025.) $)
+    rhmcomulmpl $p |- ( ph -> ( H o. ( F .x. G ) ) =
+                              ( ( H o. F ) .xb ( H o. G ) ) ) $=
+      ( vk vf vd ve cv ccnv cn cima cfn wcel cn0 cmap co crab cle cofr wbr cmin
+      cfv cof cmulr cmpt cgsu ccom cbs crh eqid rhmf syl crg rhmrcl1 rhmmpllem2
+      wf mplelf cofmpt wa cvv c0g ccmn ringcmnd adantr rhmrcl2 ringgrpd grpmndd
+      cmnd ovex rabex cmhm cghm rhmghm ghmmhm 3syl ad2antrr breq1 biimpi adantl
+      elrab simpld ffvelcdmd simplr psrbagf simprd psrbagcon syl3anc rhmmpllem1
+      a1i ringcld gsummptmhm wceq rhmmul fvco3d oveq12d eqtr4d mpteq2dva oveq2d
+      eqtr3d eqtrd mplmul coeq2d mhmcompl 3eqtr4d ) ALUEUFUIUJUKULUMUNZUFUOMUPU
+      QZURZFUGUHUIZUEUIZUSUTZVAZUHYHURZUGUIZJVCZYJYNVBVDUQZKVCZFVEVCZUQZVFVGUQZ
+      VFZVHZUEYHGUGYMYNLJVHZVCZYPLKVHZVCZGVEVCZUQZVFZVGUQZVFZLJKIUQZVHUUCUUEHUQ
+      AUUBUEYHYTLVCZVFUUKAUEYHYTFVIVCZGVIVCZLALFGVJUQUNZUUNUUOLVQUBUUNUUOFGLUUN
+      VKZUUOVKVLVMAUGUHYHFUFUEMJKYHVKZAUUPFVNUNZUBFGLVOVMZABYHDFUFMUUNJOUUQQUUR
+      UCVRZABYHDFUFMUUNKOUUQQUURUDVRZVPVSAUEYHUUMUUJAYJYHUNZVTZGUGYMYSLVCZVFZVG
+      UQUUMUUJUVDUGYMUUNYSFGLWAFWBVCZUUQUVGVKAFWCUNUVCAFUUTWDWEAGWIUNUVCAGAGAUU
+      PGVNUNUBFGLWFVMWGWHWEYMWAUNUVDYLUHYHYFUFYGUOMUPWJWKWKXJALFGWLUQUNZUVCAUUP
+      LFGWMUQUNUVHUBFGLWNFGLWOWPZWEUVDYNYMUNZVTZUUNFYRYOYQUUQYRVKZAUUSUVCUVJUUT
+      WQUVKYHUUNYNJAYHUUNJVQUVCUVJUVAWQZUVKYNYHUNZYNYJYKVAZUVJUVNUVOVTZUVDUVJUV
+      PYLUVOUHYNYHYIYNYJYKWRXAWSWTZXBZXCZUVKYHUUNYPKAYHUUNKVQUVCUVJUVBWQZUVKYPY
+      HUNZYPYJYKVAZUVKUVCMUOYNVQZUVOUWAUWBVTAUVCUVJXDUVKUVNUWCUVRYHUFYNMUURXEVM
+      UVKUVNUVOUVQXFYHUFYJYNMUURXGXHXBZXCZXKAUGUHYHFUFUEMJKUURUUTUVAUVBXIXLUVDU
+      VFUUIGVGUVDUGYMUVEUUHUVKUVEYOLVCZYQLVCZUUGUQZUUHUVKUUPYOUUNUNYQUUNUNUVEUW
+      HXMAUUPUVCUVJUBWQUVSUWEYOYQFGYRUUGLUUNUUQUVLUUGVKZXNXHUVKUUDUWFUUFUWGUUGU
+      VKYHUUNYNLJUVMUVRXOUVKYHUUNYPLKUVTUWDXOXPXQXRXSXTXRYAAUULUUALAUGUHBYHDFIY
+      RUFUEJKMOQUVLSUURUCUDYBYCAUGUHCYHEGHUUGUFUEUUCUUEMPRUWITUURABCDEFGJLMNOPQ
+      RUAUVIUCYDABCDEFGKLMNOPQRUAUVIUDYDYBYE $.
+  $}
+
+  ${
+    $d P p x y $.  $d Q x y $.  $d B x y $.  $d F x y $.  $d ph x y $.
+    $d H p $.  $d Q p $.  $d B p $.  $d ph p $.  $d H d $.  $d I d f $.
+    $d R d f $.  $d S d f $.  $d V d $.  $d ph d $.
+    rhmmpl.p $e |- P = ( I mPoly R ) $.
+    rhmmpl.q $e |- Q = ( I mPoly S ) $.
+    rhmmpl.b $e |- B = ( Base ` P ) $.
+    rhmmpl.f $e |- F = ( p e. B |-> ( H o. p ) ) $.
+    rhmmpl.i $e |- ( ph -> I e. V ) $.
+    rhmmpl.h $e |- ( ph -> H e. ( R RingHom S ) ) $.
+    $( Provide a ring homomorphism between two polynomial algebras given a ring
+       homomorphism between two structures.  Compare ~ pwsco2rhm .  TODO:
```

Comment:
Which are the "two structures"? These are the structures (must be rings!) whose base sets contain the coefficients ofthe correponding polynimials. I don't know if there is any shorter expression for that...

Reply:
Revision OK.